scripts/ios/build-sargon.sh

```diff
@@ -30,35 +36,39 @@ done
 # Potential optimizations for the future:
 # regularly check: https://github.com/ianthetechie/uniffi-starter/blob/main/rust/build-ios.sh
 # for improvements!
-#
-# * Only build one simulator arch for local development (we build both since many still use Intel Macs)
-# * Option to do debug builds instead for local development
-fat_simulator_lib_dir="target/ios-simulator-fat/release"
 
 generate_ffi() {
   echo "📦 Generating framework module mapping and FFI bindings"
-  cargo run --bin uniffi-bindgen generate --library target/aarch64-apple-ios/release/lib$1.dylib --language swift --out-dir target/uniffi-xcframework-staging
+  if $maconly; then
+    local TARGET_FOR_DYLIB_PATH="aarch64-apple-darwin"
+  else
+    local TARGET_FOR_DYLIB_PATH="aarch64-apple-ios"
+  fi 
+  cargo run --bin uniffi-bindgen generate --library target/$TARGET_FOR_DYLIB_PATH/release/lib$1.dylib --language swift --out-dir target/uniffi-xcframework-staging
   mkdir -p apple/Sources/UniFFI/
   mv target/uniffi-xcframework-staging/*.swift apple/Sources/UniFFI/
   mv target/uniffi-xcframework-staging/$1FFI.modulemap target/uniffi-xcframework-staging/module.modulemap  # Convention requires this have a specific name
 }
 
-create_fat_simulator_lib() {
-  echo "📦 Creating a fat library for x86_64 and aarch64 simulators"
```

Comment:
No more lipo - it is used to combine x86 and arm64 together for the same destination, ie intel and apple silicon mac binaries together, which we do not do.